webapp/channels/src/components/suggestion/suggestion_list.tsx

```diff
@@ -75,37 +88,38 @@ export default class SuggestionList extends React.PureComponent {
     }
 
     updateMaxHeight = () => {
-        if (!this.props.inputRef?.current) {
+        if (!this.props.inputRef) {
             return;
         }
 
-        const inputHeight = this.props.inputRef.current.clientHeight ?? 0;
+        //const inputElement = (this.props.inputRef as React.RefObject<HTMLInputElement>).current;
+        const inputHeight = (this.props.inputRef as React.RefObject<HTMLInputElement>).current!.clientHeight ?? 0;
 
         this.maxHeight = Math.min(
             window.innerHeight - (inputHeight + Constants.POST_MODAL_PADDING),
             Constants.SUGGESTION_LIST_MAXHEIGHT,
         );
 
         if (this.contentRef.current) {
-            this.contentRef.current.style['max-height'] = this.maxHeight;
+            this.contentRef.current.style.maxHeight = `${this.maxHeight}px`;
         }
     };
 
     announceLabel() {
-        const suggestionReadOut = this.props.ariaLiveRef.current;
+        const suggestionReadOut = (this.props.ariaLiveRef as React.RefObject<HTMLDivElement>).current;
```

Comment:
I think it'd be a bit better to write this with a `?.` since the if statement below would null check it. That way, we don't need to write out the type name again
```suggestion
        const suggestionReadOut = this.props.ariaLiveRef?.current;
```

Reply:
Using just `const suggestionReadOut = this.props.ariaLiveRef?.current;` I get the following error: `property 'current' does not exist on type 'RefObject<HTMLDivElement> | ((instance: HTMLDivElement | null) => void)'.
  Property 'current' does not exist on type '(instance: HTMLDivElement | null) => void'.`
  
However, using `const suggestionReadOut = (this.props.ariaLiveRef as React.RefObject<HTMLDivElement>)?.current;` still works.
  
Alternatively, we can perform the null check early on. Maybe something like this:
`
const suggestionnReadOutRef =  this.props.ariaLiveRef;
if(suggestionnReadOutRef ){
const  suggestionReadOut = (suggestionnReadOutRef  as React.RefObject<HTMLDivElement>).current
suggestionReadOut.innerHTML = this.currentLabel as string;
}
`

Reply:
Ah, okay. I thought the type assertion was just because `ariaLiveRef` could be undefined, but it's actually because that prop is typed as `React.Ref` instead of `React.RefObject`. `Ref` allows for function references which wouldn't have a `.current` field. It also looks like we use `RefObject` in most other places as well

Reply:
Ah, didn't realize the subtlety between `Ref` and `RefObject`. Thanks for pointing that out